### PR TITLE
Make sure requirements are included in sdist

### PR DESCRIPTION
**Pull request by jrapin**
#1619

### CODE REVIEW

**Comment by gwenzek** [comment]
I don't see how this is related to the issue ?
Isn't the problem a setuptools version in which case adding setuptools to our requirements/dev.txt should be enough ?

If we really need to add files to the distributed package, could we use add them in setup.py to the package_data entry ?

**Comment by gwenzek** [comment]
Also wouldn't switching from sdist to bdist_wheel would prevent this kind of issue ?

**Comment by jrapin** [comment]
> Also wouldn't switching from sdist to bdist_wheel would prevent this kind of issue ?

Some people prefer sdist so I would rather it worked as wel
I reran with yet another version and it seems it worked this time, either my version was too old (I thought I reinstalled it though) or there was a temporary bug

**Comment by jrapin** [comment]
@gwenzek 
I doooooont get it
works fine on my Mac, but in the CI here is the list of generated files, which... does not include `requirements/main.txt` even though this is the exact same setuptools version `56.2.0`. Either there is a subdependency we are missing, or there is a difference between platforms/python versions :s 

```
ting submitit.egg-info', 'writing submitit.egg-info/PKG-INFO', 'writing dependency_links to submitit.egg-info/dependency_links.txt', 'writing requirements to submitit.egg-info/requires.txt', 'writing top-level names to submitit.egg-info/top_level.txt', "writing manifest file 'submitit.egg-info/SOURCES.txt'", "adding license file 'LICENSE' (matched pattern 'LICEN[CS]E*')", "reading manifest file 'submitit.egg-info/SOURCES.txt'", "writing manifest file 'submitit.egg-info/SOURCES.txt'", 'running check', 'creating submitit-1.3.3', 'creating submitit-1.3.3/submitit', 'creating submitit-1.3.3/submitit.egg-info', 'creating submitit-1.3.3/submitit/auto', 'creating submitit-1.3.3/submitit/core', 'creating submitit-1.3.3/submitit/local', 'creating submitit-1.3.3/submitit/slurm', 'copying files to submitit-1.3.3...', 'copying LICENSE -> submitit-1.3.3', 'copying README.md -> submitit-1.3.3', 'copying pyproject.toml -> submitit-1.3.3', 'copying setup.py -> submitit-1.3.3', 'copying submitit/__init__.py -> submitit-1.3.3/submitit', 'copying submitit/conftest.py -> submitit-1.3.3/submitit', 'copying submitit/helpers.py -> submitit-1.3.3/submitit', 'copying submitit/py.typed -> submitit-1.3.3/submitit', 'copying submitit/test_documentation.py -> submitit-1.3.3/submitit', 'copying submitit/test_helpers.py -> submitit-1.3.3/submitit', 'copying submitit/test_pickle.py -> submitit-1.3.3/submitit', 'copying submitit.egg-info/PKG-INFO -> submitit-1.3.3/submitit.egg-info', 'copying submitit.egg-info/SOURCES.txt -> submitit-1.3.3/submitit.egg-info', 'copying submitit.egg-info/dependency_links.txt -> submitit-1.3.3/submitit.egg-info', 'copying submitit.egg-info/not-zip-safe -> submitit-1.3.3/submitit.egg-info', 'copying submitit.egg-info/requires.txt -> submitit-1.3.3/submitit.egg-info', 'copying submitit.egg-info/top_level.txt -> submitit-1.3.3/submitit.egg-info', 'copying submitit/auto/__init__.py -> submitit-1.3.3/submitit/auto', 'copying submitit/auto/auto.py -> submitit-1.3.3/submitit/auto', 'copying submitit/auto/test_auto.py -> submitit-1.3.3/submitit/auto', 'copying submitit/core/__init__.py -> submitit-1.3.3/submitit/core', 'copying submitit/core/_submit.py -> submitit-1.3.3/submitit/core', 'copying submitit/core/core.py -> submitit-1.3.3/submitit/core', 'copying submitit/core/job_environment.py -> submitit-1.3.3/submitit/core', 'copying submitit/core/logger.py -> submitit-1.3.3/submitit/core', 'copying submitit/core/plugins.py -> submitit-1.3.3/submitit/core', 'copying submitit/core/submission.py -> submitit-1.3.3/submitit/core', 'copying submitit/core/test_core.py -> submitit-1.3.3/submitit/core', 'copying submitit/core/test_plugins.py -> submitit-1.3.3/submitit/core', 'copying submitit/core/test_utils.py -> submitit-1.3.3/submitit/core', 'copying submitit/core/utils.py -> submitit-1.3.3/submitit/core', 'copying submitit/local/__init__.py -> submitit-1.3.3/submitit/local', 'copying submitit/local/_local.py -> submitit-1.3.3/submitit/local', 'copying submitit/local/debug.py -> submitit-1.3.3/submitit/local', 'copying submitit/local/local.py -> submitit-1.3.3/submitit/local', 'copying submitit/local/test_debug.py -> submitit-1.3.3/submitit/local', 'copying submitit/local/test_local.py -> submitit-1.3.3/submitit/local', 'copying submitit/slurm/__init__.py -> submitit-1.3.3/submitit/slurm', 'copying submitit/slurm/slurm.py -> submitit-1.3.3/submitit/slurm', 'copying submitit/slurm/test_slurm.py -> submitit-1.3.3/submitit/slurm', 'Writing submitit-1.3.3/setup.cfg', 'creating dist', 'Creating tar archive', "removing 'submitit-1.3.3' (and everything under it)"]
```

File: .circleci/config.yml [hunk 1]
```diff
@@ -35,6 +35,20 @@ jobs:
...
+      - checkout
+      - attach_workspace:
+          at: venv
+      - run: |
+          pip install -U setuptools
+          rm -rf dist  # make sure it's clean
+          python setup.py sdist > log.txt
+          python -c "from pathlib import Path;lines = Path('log.txt').read_text().splitlines(); assert any('copying LICENSE' in x for x in lines), lines"
+          python -c "from pathlib import Path;lines = Path('log.txt').read_text().splitlines(); assert any('copying requirements/main.txt' in x for x in lines), lines"
+
```
Comment (jrapin) [inline .circleci/config.yml:51]: @gwenzek might be worth writting this to the MAKEFILE somewhere? Not very familiar with how to make it work though
(and yes, it's ugly :s)

**Pull request closed without merging**